// vite_react_shadcn_ts/src/components/AnimatedContainer.tsx

import React, { useEffect, useState } from "react";
import { cn } from "@/lib/utils";

interface AnimatedContainerProps {
  children: React.ReactNode;
  className?: string;
  delay?: number;
  animationType?: "fade" | "slide-up" | "slide-down" | "scale" | "none";
}

const AnimatedContainer: React.FC<AnimatedContainerProps> = ({
  children,
  className,
  delay = 0,
  animationType = "fade",
}) => {
  const [isVisible, setIsVisible] = useState(false);

  useEffect(() => {
    const timer = setTimeout(() => {
      setIsVisible(true);
    }, delay);

    return () => clearTimeout(timer);
  }, [delay]);

  const getAnimationClass = () => {
    if (!isVisible) return "opacity-0";
    
    switch (animationType) {
      case "fade":
        return "animate-fade-in";
      case "slide-up":
        return "animate-slide-up";
      case "slide-down":
        return "animate-slide-down";
      case "scale":
        return "animate-scale-in";
      case "none":
        return "opacity-100";
      default:
        return "animate-fade-in";
    }
  };

  return (
    <div
      className={cn(
        getAnimationClass(),
        "transition-all duration-500",
        className
      )}
      style={{
        animationDelay: `${delay}ms`,
        animationFillMode: "forwards",
      }}
    >
      {children}
    </div>
  );
};

export default AnimatedContainer;
