// vite_react_shadcn_ts/src/components/WasteCard.tsx

import React from "react";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { cn } from "@/lib/utils";
import AnimatedContainer from "./AnimatedContainer";

interface WasteCardProps {
  title: string;
  color: string;
  description?: string;
  icon?: React.ReactNode;
  className?: string;
  onClick?: () => void;
  delay?: number;
  children?: React.ReactNode;
}

const WasteCard: React.FC<WasteCardProps> = ({
  title,
  color,
  description,
  icon,
  className,
  onClick,
  delay = 0,
  children,
}) => {
  return (
    <AnimatedContainer delay={delay} animationType="scale">
      <Card
        className={cn(
          "overflow-hidden transition-all duration-300 border border-border/50",
          onClick ? "hover:shadow-md cursor-pointer" : "",
          className
        )}
        onClick={onClick}
      >
        <div
          className={cn(
            "h-2 w-full",
            `bg-waste-${color}`
          )}
        />
        <CardHeader className="p-4 pb-0">
          <CardTitle className="text-lg font-medium flex items-center gap-2">
            {icon && <span className="text-waste-${color}">{icon}</span>}
            {title}
          </CardTitle>
        </CardHeader>
        <CardContent className="p-4 pt-2">
          {description && <p className="text-sm text-muted-foreground">{description}</p>}
          {children}
        </CardContent>
      </Card>
    </AnimatedContainer>
  );
};

export default WasteCard;
