// vite_react_shadcn_ts/src/components/WasteResult.tsx

import React from "react";
import { Check, Info } from "lucide-react";
import { Card, CardContent } from "@/components/ui/card";
import { Progress } from "@/components/ui/progress";
import AnimatedContainer from "./AnimatedContainer";
import { WasteResult as WasteResultType } from "@/utils/wasteAnalyzer";

interface WasteResultProps {
  result: WasteResultType;
  className?: string;
}

const WasteResult: React.FC<WasteResultProps> = ({ result, className }) => {
  const { category, confidence, instructions, tips } = result;
  
  // Format confidence as percentage
  const confidencePercent = Math.round(confidence * 100);
  
  // Get category display name with first letter capitalized
  const categoryName = category.charAt(0).toUpperCase() + category.slice(1);
  
  // Choose color based on category
  const categoryColors: Record<string, string> = {
    paper: "bg-waste-paper text-waste-paper",
    plastic: "bg-waste-plastic text-waste-plastic",
    glass: "bg-waste-glass text-waste-glass",
    metal: "bg-waste-metal text-waste-metal",
    organic: "bg-waste-organic text-waste-organic",
    landfill: "bg-waste-landfill text-waste-landfill",
  };
  
  const colorClass = categoryColors[category] || "bg-primary text-primary";
  const bgColorClass = colorClass.split(" ")[0];
  const textColorClass = colorClass.split(" ")[1];

  return (
    <div className={className}>
      <AnimatedContainer animationType="scale">
        <Card className="overflow-hidden border border-border/50">
          <div className={`h-2 w-full ${bgColorClass}`} />
          <CardContent className="p-6">
            <div className="flex flex-col gap-4">
              <div>
                <div className="flex justify-between items-center mb-3">
                  <div className="flex items-center gap-2">
                    <span className={`h-3 w-3 rounded-full ${bgColorClass}`} />
                    <h3 className="font-medium">Identified as {categoryName}</h3>
                  </div>
                  <div className="text-sm text-muted-foreground">
                    Confidence: {confidencePercent}%
                  </div>
                </div>
                <Progress value={confidencePercent} className={bgColorClass} />
              </div>

              <div>
                <h4 className="text-sm font-medium mb-2 flex items-center gap-1">
                  <Check className="h-4 w-4" /> Instructions
                </h4>
                <ul className="space-y-2">
                  {instructions.map((instruction, i) => (
                    <AnimatedContainer
                      key={i}
                      delay={i * 100}
                      animationType="slide-up"
                    >
                      <li className="flex items-start gap-2">
                        <span className="flex-shrink-0 h-5 w-5 rounded-full bg-primary/10 flex items-center justify-center text-xs text-primary">
                          {i + 1}
                        </span>
                        <span className="text-sm">{instruction}</span>
                      </li>
                    </AnimatedContainer>
                  ))}
                </ul>
              </div>

              <div className="bg-secondary/50 p-3 rounded-lg mt-2">
                <h4 className="text-sm font-medium mb-2 flex items-center gap-1">
                  <Info className="h-4 w-4" /> Helpful Tips
                </h4>
                <ul className="space-y-2">
                  {tips.map((tip, i) => (
                    <AnimatedContainer
                      key={i}
                      delay={200 + i * 100}
                      animationType="fade"
                    >
                      <li className="text-sm text-muted-foreground">{tip}</li>
                    </AnimatedContainer>
                  ))}
                </ul>
              </div>
            </div>
          </CardContent>
        </Card>
      </AnimatedContainer>
    </div>
  );
};

export default WasteResult;
