// vite_react_shadcn_ts/src/utils/wasteAnalyzer.ts

// A simple mock analyzer that returns predefined waste categories
// In a real app, this would be replaced with actual image recognition model

export type WasteCategory = 
  | 'paper'
  | 'plastic'
  | 'glass'
  | 'metal'
  | 'organic'
  | 'landfill';

export interface WasteResult {
  category: WasteCategory;
  confidence: number;
  instructions: string[];
  tips: string[];
}

const wasteInstructions: Record<WasteCategory, string[]> = {
  paper: [
    "Remove any plastic or metal parts",
    "Flatten cardboard boxes",
    "Place in paper recycling bin"
  ],
  plastic: [
    "Empty and rinse container",
    "Replace cap if recyclable",
    "Check recycling number and local guidelines",
    "Place in plastic recycling bin"
  ],
  glass: [
    "Empty and rinse container",
    "Remove lids or caps",
    "Check for any non-glass parts",
    "Place in glass recycling bin"
  ],
  metal: [
    "Empty and rinse container",
    "Remove any non-metal parts",
    "Flatten if possible (for cans)",
    "Place in metal recycling bin"
  ],
  organic: [
    "Remove any non-compostable items",
    "Cut larger items into smaller pieces",
    "Place in compost bin or green waste"
  ],
  landfill: [
    "Make sure item cannot be recycled",
    "Place in general waste bin",
    "Consider if future purchases could be recyclable"
  ]
};

const wasteTips: Record<WasteCategory, string[]> = {
  paper: [
    "Paper can be recycled 5-7 times before fibers become too short",
    "Receipts and thermal paper are not recyclable due to chemical coatings"
  ],
  plastic: [
    "Plastic numbers 1 and 2 are most widely recyclable",
    "Plastic bags and films often need special collection"
  ],
  glass: [
    "Glass is 100% recyclable and can be recycled indefinitely",
    "Different colors of glass should be separated when possible"
  ],
  metal: [
    "Aluminum cans can be recycled indefinitely",
    "Recycling aluminum uses 95% less energy than producing new aluminum"
  ],
  organic: [
    "Composting reduces methane emissions from landfills",
    "Compost creates nutrient-rich soil for plants"
  ],
  landfill: [
    "Try to reduce consumption of non-recyclable items",
    "Consider switching to reusable alternatives"
  ]
};

// In a real app, this would analyze the image using a trained model
export const analyzeWasteImage = async (imageFile: File): Promise<WasteResult> => {
  // Simulate API call delay
  await new Promise(resolve => setTimeout(resolve, 1500));
  
  // For demo purposes, randomly select a waste category
  const categories: WasteCategory[] = ['paper', 'plastic', 'glass', 'metal', 'organic', 'landfill'];
  const randomCategory = categories[Math.floor(Math.random() * categories.length)];
  
  // Random confidence level between 70% and 95%
  const confidence = 0.7 + (Math.random() * 0.25);
  
  return {
    category: randomCategory,
    confidence: confidence,
    instructions: wasteInstructions[randomCategory],
    tips: wasteTips[randomCategory]
  };
};
